### PR TITLE
Corne ish zen v2

### PR DESCRIPTION
**Pull request by LOWPROKB**
## Board/Shield Check-list
 - [ ✓] This board/shield is tested working on real hardware
 - [ ✓] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [✓ ] `.zmk.yml` metadata file added
 - [✓ ] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [ ✓] General consistent formatting of DeviceTree files
 - [ ✓] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [ ✓] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [✓ ] If split, no name added for the right/peripheral half
 - [ ✓] Kconfig.defconfig file correctly wraps *all* configuration in conditional on the shield symbol
 - [ ✓] `.conf` file has optional extra features commented out
 - [✓ ] Keyboard/PCB is part of a shipped group buy or is generally available in stock to purchase (OSH/personal projects without general availability should create a zmk-config repo instead)

### CODE REVIEW

File: app/boards/arm/corneish_zen/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,60 @@
+set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/CMakeLists.txt:1]: Oh! Old checkouts might still have this, but this got deprecated in the Zephyr 3.0 upgrade. See https://zmk.dev/blog/2022/04/02/zephyr-3-0#uf2-builds for how to do this the right way now.

Comment (LOWPROKB) [reply app/boards/arm/corneish_zen/CMakeLists.txt]: Yes, that was it. I had the new parts in the config, but didn't remove the old parts. building fine now

**Comment by manna-harbour** [comment]
It builds for me as a zmk-config board with ZMK main: https://github.com/manna-harbour/miryoku_zmk/actions/runs/3278634739.

File: .vscode/c_cpp_properties.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (manna-harbour) [inline .vscode/c_cpp_properties.json:1]: Accidentally included?

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,7 +1,12 @@
 {
```
Comment (manna-harbour) [inline .vscode/settings.json:1]: Accidentally included?

**Review by manna-harbour — CHANGES_REQUESTED** [review]

**Comment by manna-harbour** [comment]
A uf2 is being generated for left only.

**Comment by LOWPROKB** [comment]
Artifacts are being produced by the left and right builds now

File: app/boards/arm/corneish_zen/Kconfig.defconfig [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+	default ZMK_DISPLAY_STATUS_SCREEN_CUSTOM
+endchoice
+
+if ZMK_DISPLAY
+
+config SPI
+	default y
+
+config IL0323
+	bool
```
[the remark this answers is not in the document]
Comment (LOWPROKB) [reply app/boards/arm/corneish_zen/Kconfig.defconfig]: tried removing the cast statement but did not appear to have an effect on the build

**Comment by manna-harbour** [comment]
> Artifacts are being produced by the left and right builds now

Thanks, looks good now: https://github.com/manna-harbour/miryoku_zmk/actions/runs/3285476237.

File: app/boards/arm/corneish_zen/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  zephyr_library_sources(widgets/icons/bluetooth_connected_2.c)
+  zephyr_library_sources(widgets/icons/bluetooth_connected_3.c)
+  zephyr_library_sources(widgets/icons/bluetooth_connected_4.c)
+  zephyr_library_sources(widgets/icons/bluetooth_connected_5.c)
+  zephyr_library_sources(widgets/icons/bluetooth_advertising_1.c)
+  zephyr_library_sources(widgets/icons/bluetooth_advertising_2.c)
+  zephyr_library_sources(widgets/icons/bluetooth_advertising_3.c)
+  zephyr_library_sources(widgets/icons/bluetooth_advertising_4.c)
+  zephyr_library_sources(widgets/icons/bluetooth_advertising_5.c)
+endif()
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/CMakeLists.txt:44]: These should probably be conditional based on the config for each widget that uses/"owns" a given icon file.

File: app/boards/arm/corneish_zen/Kconfig.board [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#
+# Copyright (c) 2021 Darryl deHaan
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/Kconfig.board:2]:
```suggestion
# Copyright (c) 2022 Darryl deHaan
```

File: app/boards/arm/corneish_zen/Kconfig.defconfig [hunk 7]
```diff
@@ -0,0 +1,119 @@
+#
+# Copyright (c) 2021 Darryl deHaan
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/Kconfig.defconfig:2]:
```suggestion
# Copyright (c) 2022 Darryl deHaan
```

File: app/boards/arm/corneish_zen/Kconfig.defconfig [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+# SPDX-License-Identifier: MIT
+#
+
+if BOARD_CORNEISH_ZEN_V2_LEFT
+
+	config ZMK_KEYBOARD_NAME
+	default "Corne-ish Zen"
+
+	config ZMK_SPLIT_ROLE_CENTRAL
+	default y
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/Kconfig.defconfig:12]:
```suggestion
config ZMK_KEYBOARD_NAME
	default "Corne-ish Zen"

config ZMK_SPLIT_ROLE_CENTRAL
	default y
```

File: app/boards/arm/corneish_zen/Kconfig.defconfig [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+endif # BOARD_CORNEISH_ZEN_V2_LEFT
+
+if BOARD_CORNEISH_ZEN_V2_RIGHT
+
+	config ZMK_KEYBOARD_NAME
+	default "Corne-ish Zen rt"
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/Kconfig.defconfig:19]: We dont typically add names for the right side any more... it avoids them being visible to hosts, so folks don't try to pair to them.

File: app/boards/arm/corneish_zen/Kconfig.defconfig [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+#menuconfig SENSOR
+#    bool "Sensor Drivers"
+
+#config BQ274XX
+#    bool "BQ274xx Fuel Gauge"
+#    depends on I2C && SENSOR
+
+#config SENSOR_CAN_FETCH_ALL
+#	bool "Can the fuel gauge fetch all channels at once"
+#	default y
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/Kconfig.defconfig:50]:
```suggestion
```

File: app/boards/arm/corneish_zen/Kconfig.defconfig [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+config SPI
+    bool
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/Kconfig.defconfig:53]:
```suggestion
```

File: app/boards/arm/corneish_zen/Kconfig.defconfig [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+	select LVGL_USE_IMG
+
+choice ZMK_DISPLAY_STATUS_SCREEN
+	default ZMK_DISPLAY_STATUS_SCREEN_CUSTOM
+endchoice
+
+if ZMK_DISPLAY
+
+config SPI
+	default y
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/Kconfig.defconfig:90]: Already defaulted earlier... If SPI only needed w/ displays, remove the other default earlier in the file.

File: app/boards/arm/corneish_zen/Kconfig.defconfig [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+
+config IL0323
+	default y
+
+config ZMK_DISPLAY_BLANK_ON_IDLE
+	default n
+
+#config DISPLAY_WOB
+#	bool "White on Black Display"
+#	default n
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/Kconfig.defconfig:100]:
```suggestion
```

File: app/boards/arm/corneish_zen/Kconfig.defconfig [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+
+menuconfig CUSTOM_WIDGET_LAYER_STATUS
+	bool "custom layer status widget"
+
+menuconfig CUSTOM_WIDGET_PERIPHERAL_STATUS
+	bool "custom peripheral status widget"
+
+# config CORNE_ISH_ZEN_20
+#    bool
+
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/Kconfig.defconfig:118]:
```suggestion
```

File: app/boards/arm/corneish_zen/corneish_zen.dtsi [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+
+&uart0 {
+	compatible = "nordic,nrf-uarte";
+	status = "okay";
+	current-speed = <115200>;
+	tx-pin = <39>;
+	rx-pin = <44>;
+	rts-pin = <34>;
+	cts-pin = <9>;
+};
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen.dtsi:98]: Do you actually use the UART for anything? If not, ditch this setup:

```suggestion
```

File: app/boards/arm/corneish_zen/corneish_zen.dtsi [hunk 16]
```diff
@@ -0,0 +1,159 @@
...
+
+&rcc {
+  clocks = <&pll>;
+  clock-frequency = <DT_FREQ_M(72)>;
+  ahb-prescaler = <1>;
+  apb1-prescaler = <2>;
+}; */
+
+
+
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen.dtsi:119]: Commented out. Nuke it?

```suggestion
```

File: app/boards/arm/corneish_zen/corneish_zen_v2_left.dts [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+		compatible = "ti,bq274xx";
+		label = "BATTERY";
+		reg = <0x55>;
+		design-voltage = <3700>; //Battery Design Volatge in mV
+		design-capacity = <180>; //Battery Design Capacity in mAh
+		taper-current = <2>; //Battery Taper current in mAh
+		terminate-voltage = <2750>; //Battery Terminate Voltage in mV
+	};
+}; */
+
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_left.dts:76]:
```suggestion
```

File: app/boards/arm/corneish_zen/corneish_zen_v2_left.dts [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+		reg =  <0>;
+		label = "DISPLAY";
+		width = <80>;
+		height = <128>;
+		spi-max-frequency = <4000000>;
+		dc-gpios = <&gpio0 11 GPIO_ACTIVE_LOW>;
+		busy-gpios = <&gpio0 25 GPIO_ACTIVE_LOW>;
+		reset-gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+		pwr = [03 00 26 26];
+		//softstart = [17 17 17 17];
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_left.dts:96]:
```suggestion
```

Remove?

File: app/boards/arm/corneish_zen/corneish_zen_v2_left_defconfig [hunk 19]
```diff
@@ -0,0 +1,94 @@
...
+#CONFIG_LOG_STRDUP_BUF_COUNT=160
+#CONFIG_I2C_LOG_LEVEL_DBG=y
+#CONFIG_SPI_LOG_LEVEL_DBG=y
+#CONFIG_DISPLAY_LOG_LEVEL_DBG=y
+#CONFIG_LVGL_LOG_LEVEL_DBG=y
+#CONFIG_LVGL_USE_DEBUG=y
+#CONFIG_SENSOR_LOG_LEVEL_DBG=y
+
+# Turn on USB CDC ACM device
+CONFIG_ZMK_USB=y
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_left_defconfig:80]: This is already defaulted in the other Kconfig file, it seems.

File: app/boards/arm/corneish_zen/corneish_zen_v2_right_defconfig [hunk 20]
```diff
@@ -0,0 +1,91 @@
...
+#CONFIG_LOG_STRDUP_BUF_COUNT=60
+#CONFIG_I2C_LOG_LEVEL_DBG=y
+#CONFIG_SPI_LOG_LEVEL_DBG=y
+#CONFIG_DISPLAY_LOG_LEVEL_DBG=y
+#CONFIG_LVGL_LOG_LEVEL_DBG=y
+#CONFIG_LVGL_USE_DEBUG=y
+#CONFIG_SENSOR_LOG_LEVEL_DBG=y
+
+# Turn on USB CDC ACM device
+CONFIG_ZMK_USB=y
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_right_defconfig:79]: Ditto

File: app/package.json [hunk 21]
```diff
@@ -18,6 +18,6 @@
   },
   "homepage": "https://zmk.dev/",
   "devDependencies": {
-    "prettier": "^2.4.0"
+    "prettier": "^2.7.1"
```
Comment (petejohanson) [inline app/package.json:21]: Please revert these unrelated changes.

**Review by petejohanson — CHANGES_REQUESTED** [review]
An assortment of items to address. Thanks!

File: app/boards/arm/corneish_zen/CMakeLists.txt (hunk 5 above)
Comment (caksoylar) [reply app/boards/arm/corneish_zen/CMakeLists.txt:44]: Here is my attempt at this, if it helps: https://github.com/caksoylar/zmk/commit/dbdebb053f1320d0aad5638c6b4490ab90e516cc

File: app/package.json (hunk 21 above)
Comment (LOWPROKB) [reply app/package.json:21]: strange... my local version says only 2.7.1

File: .vscode/settings.json [hunk 22]
```diff
@@ -1,7 +0,0 @@
-{
```
Comment (petejohanson) [inline .vscode/settings.json:1]: Please rebase and restore this file so it's untouched in the PR.

File: app/boards/arm/corneish_zen/CMakeLists.txt [hunk 23]
```diff
@@ -0,0 +1,62 @@
+if(CONFIG_ZMK_DISPLAY)
+  target_sources_ifdef(CONFIG_CUSTOM_WIDGET_BATTERY_STATUS app PRIVATE widgets/battery_status.c)
+  target_sources_ifdef(CONFIG_CUSTOM_WIDGET_OUTPUT_STATUS app PRIVATE widgets/output_status.c)
+  target_sources_ifdef(CONFIG_CUSTOM_WIDGET_LAYER_STATUS app PRIVATE widgets/layer_status.c)
+  target_sources_ifdef(CONFIG_CUSTOM_WIDGET_PERIPHERAL_STATUS app PRIVATE widgets/peripheral_status.c)
+  # target_sources_ifdef(CONFIG_ZMK_WIDGET_WPM_STATUS app PRIVATE wpm_status.c)
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/CMakeLists.txt:6]:
```suggestion
```

File: app/boards/arm/corneish_zen/CMakeLists.txt [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+
+  add_subdirectory_ifdef(CONFIG_ZMK_DISPLAY_STATUS_SCREEN_CUSTOM  widgets/icons)
+endif()
+
+zephyr_library()
+
+# zephyr_library_sources(corne-ish_zen.c)
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/CMakeLists.txt:13]:
```suggestion
```

File: app/boards/arm/corneish_zen/Kconfig.defconfig [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+# SPDX-License-Identifier: MIT
+#
+
+if BOARD_CORNEISH_ZEN_V2_LEFT
+
+	config ZMK_KEYBOARD_NAME
+		default "Corne-ish Zen"
+
+	config ZMK_SPLIT_ROLE_CENTRAL
+		default y
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/Kconfig.defconfig:12]:
```suggestion
config ZMK_KEYBOARD_NAME
	default "Corne-ish Zen"

config ZMK_SPLIT_ROLE_CENTRAL
	default y
```

File: app/boards/arm/corneish_zen/corneish_zen.dtsi [hunk 26]
```diff
@@ -0,0 +1,128 @@
...
+	model = "corneish_zen_v2";
+	compatible = "corneish_zen_v2";
+
+	chosen {
+		zephyr,code-partition = &code_partition;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zmk,kscan = &kscan0;
+		zmk,matrix_transform = &default_transform;
+		//zmk,matrix_transform = &five_column_transform;
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen.dtsi:23]: Probably better left as a comment in the keymap if you want to leave that as an example in a comment.

File: app/boards/arm/corneish_zen/corneish_zen_v2_left.keymap [hunk 27]
```diff
@@ -0,0 +1,8 @@
+/*
+*
+* Copyright (c) 2021 Darryl deHaan
+* SPDX-License-Identifier: MIT
+*
+*/
+
+#include "corneish_zen.keymap"
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_left.keymap:8]: I believe the left/right keymap files are unneeded now after some build fixes. Can you please test and if so, unify this?

File: app/boards/arm/corneish_zen/corneish_zen_v2_right.keymap [hunk 28]
```diff
@@ -0,0 +1,8 @@
+/*
+*
+* Copyright (c) 2021 Darryl deHaan
+* SPDX-License-Identifier: MIT
+*
+*/
+
+#include "corneish_zen.keymap"
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_right.keymap:8]: Ditto.

File: app/boards/arm/corneish_zen/corneish_zen_v2_right_defconfig [hunk 29]
```diff
@@ -0,0 +1,90 @@
...
+#CONFIG_SPI_LOG_LEVEL_DBG=y
+#CONFIG_DISPLAY_LOG_LEVEL_DBG=y
+#CONFIG_LVGL_LOG_LEVEL_DBG=y
+#CONFIG_LVGL_USE_DEBUG=y
+#CONFIG_SENSOR_LOG_LEVEL_DBG=y
+
+# Turn on USB CDC ACM device
+CONFIG_USB_DEVICE_STACK=y
+CONFIG_USB_CDC_ACM=y
+CONFIG_USB_CDC_ACM_RINGBUF_SIZE=1024
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_right_defconfig:81]: These usually aren't enabled by default, why here?

File: app/boards/arm/corneish_zen/widgets/CMakeLists.txt [hunk 30]
```diff
@@ -0,0 +1,7 @@
+#
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/widgets/CMakeLists.txt:1]: Is this file actually used/needed/ If not, should be removed.

File: app/package.json (hunk 21 above)
Comment (petejohanson) [reply app/package.json:21]: :shrug: Should be able to use `git log --stat` to find which commit changed it, and fix it up w/ an interactive rebase.

**Review by petejohanson — CHANGES_REQUESTED** [review]
A few more items that need cleaning up.

Thanks!

File: .vscode/settings.json [hunk 31]
```diff
@@ -1,7 +1,8 @@
 {
   "files.associations": {
     "*.overlay": "dts",
-    "*.keymap": "dts"
+    "*.keymap": "dts",
```
Comment (petejohanson) [inline .vscode/settings.json:4]: Still some changes here that shouldn't be.

File: app/boards/arm/corneish_zen/corneish_zen.dtsi (hunk 26 above)
Comment (LOWPROKB) [reply app/boards/arm/corneish_zen/corneish_zen.dtsi:23]: This transform line is not referenced from the keymap file though... maybe I am not understanding.

File: app/boards/arm/corneish_zen/corneish_zen_v2_right_defconfig (hunk 29 above)
Comment (LOWPROKB) [reply app/boards/arm/corneish_zen/corneish_zen_v2_right_defconfig:81]: I was likely trying to solve an issue by trying everything I could think of. Will delete

File: app/boards/arm/corneish_zen/corneish_zen_v2_left_defconfig [hunk 32]
```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (c) 2022 Darryl deHaan
+# SPDX-License-Identifier: MIT
+#
+
+CONFIG_SOC_SERIES_NRF52X=y
+CONFIG_SOC_NRF52840_QIAA=y
+CONFIG_BOARD_CORNEISH_ZEN_V2_LEFT=y
+CONFIG_ZMK_SPLIT=y
+CONFIG_ZMK_SLEEP=y
```
Comment (caksoylar) [inline app/boards/arm/corneish_zen/corneish_zen_v2_left_defconfig:10]: Should deep sleep be enabled by default? I don't think any other board/shield sets that in ZMK at the moment.

Comment (petejohanson) [reply app/boards/arm/corneish_zen/corneish_zen_v2_left_defconfig:10]: We don't have a firm policy here. If a specific *vendor* board wants to enable this, I don't see a problem with it.

Thoughts?

Comment (caksoylar) [reply app/boards/arm/corneish_zen/corneish_zen_v2_left_defconfig:10]: I am in general slightly opposed, since it can cause confusion to the users if they don't know that deep sleep is enabled. If they have to enable it in the conf themselves they will at least be aware of it. However I do see the logic in letting the vendor/designer decide for their own keyboards.

Comment (LOWPROKB) [reply app/boards/arm/corneish_zen/corneish_zen_v2_left_defconfig:10]: My main concern was folks who are firmware unaware. At least they would get the battery savings.

File: app/boards/arm/corneish_zen/corneish_zen.dtsi (hunk 26 above)
Comment (LOWPROKB) [reply app/boards/arm/corneish_zen/corneish_zen.dtsi]: figured it out... just moved everything over to the keymap

**Comment by caksoylar** [comment]
Worth adding a `corneish_zen.conf` file with the contents of https://github.com/LOWPROKB/zmk-config-zen-2/blob/main/config/corneish_zen.conf? This way going through the setup script can give you the same files as the linked config repo.

File: app/boards/arm/corneish_zen/corneish_zen_v2_left.dts [hunk 33]
```diff
@@ -0,0 +1,81 @@
...
+* Copyright (c) 2021 Darryl deHaan
+* SPDX-License-Identifier: MIT
+*
+*/
+
+#include "corneish_zen.dtsi"
+
+/{
+	chosen {
+		zephyr,display = &epd;
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_left.dts:12]:
```suggestion
		zephyr,display = &epd;
		zmk,battery = &vbatt;
```

File: app/boards/arm/corneish_zen/corneish_zen_v2_left.dts [hunk 34]
```diff
@@ -0,0 +1,81 @@
...
+		blue_led: led_0 {
+			gpios = <&gpio0 16 GPIO_ACTIVE_HIGH>;
+			label = "Blue LED";
+		};
+	};
+
+    chosen {
+		zmk,battery = &vbatt;
+	};
+
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_left.dts:48]:
```suggestion
```

File: app/boards/arm/corneish_zen/corneish_zen_v2_left.dts [hunk 35]
```diff
@@ -0,0 +1,81 @@
...
+    vbatt: vbatt {
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_left.dts:49]:
```suggestion
	vbatt: vbatt {
```

File: app/boards/arm/corneish_zen/corneish_zen_v2_right.dts [hunk 36]
```diff
@@ -0,0 +1,110 @@
...
+* Copyright (c) 2021 Darryl deHaan
+* SPDX-License-Identifier: MIT
+*
+*/
+
+#include "corneish_zen.dtsi"
+
+/{
+	chosen {
+		zephyr,display = &epd;
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_right.dts:12]:
```suggestion
		zephyr,display = &epd;
		zmk,battery = &vbatt;
```

File: app/boards/arm/corneish_zen/corneish_zen_v2_right.dts [hunk 37]
```diff
@@ -0,0 +1,110 @@
...
+			compatible = "gpio-leds";
+			blue_led: led_0 {
+				gpios = <&gpio0 26 GPIO_ACTIVE_HIGH>;
+				label = "Blue LED";
+			};
+		};
+	chosen {
+		zmk,battery = &vbatt;
+	};
+
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_right.dts:48]:
```suggestion
```

File: app/boards/arm/corneish_zen/corneish_zen_v2_right.dts [hunk 38]
```diff
@@ -0,0 +1,110 @@
...
+    vbatt: vbatt {
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_right.dts:49]:
```suggestion
	vbatt: vbatt {
```

File: app/boards/arm/corneish_zen/corneish_zen_v2_right.dts [hunk 39]
```diff
@@ -0,0 +1,110 @@
...
+		label = "BATTERY";
+		io-channels = <&adc 0>;
+		output-ohms = <1960000>;
+		full-ohms = <(1960000 + 810000)>;
+	};
+};
+
+&default_transform {
+		col-offset = <6>;
+	};
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_right.dts:60]:
```suggestion
	col-offset = <6>;
};

```

File: app/boards/arm/corneish_zen/corneish_zen_v2_right.dts [hunk 40]
```diff
@@ -0,0 +1,110 @@
...
+		compatible = "ti,bq274xx";
+		label = "BATTERY";
+		reg = <0x55>;
+		design-voltage = <3700>; //Battery Design Volatge in mV
+		design-capacity = <180>; //Battery Design Capacity in mAh
+		taper-current = <2>; //Battery Taper current in mAh 2.1
+		terminate-voltage = <2750>; //Battery Terminate Voltage in mV
+	};
+}; */
+
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_right.dts:86]:
```suggestion
```

File: app/boards/arm/corneish_zen/corneish_zen_v2_right_defconfig [hunk 41]
```diff
@@ -0,0 +1,85 @@
...
+CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL=y
+CONFIG_CLOCK_CONTROL_NRF_K32SRC_30PPM=y
+
+# enable display drivers
+CONFIG_ZMK_DISPLAY_WORK_QUEUE_DEDICATED=y
+CONFIG_ZMK_DISPLAY_DEDICATED_THREAD_STACK_SIZE=2048
+CONFIG_SSD1306=n
+CONFIG_IL0323=y
+CONFIG_LVGL_HOR_RES_MAX=80
+CONFIG_LVGL_VER_RES_MAX=128
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_right_defconfig:43]:
```suggestion
```

Zephyr automatically sets this now from the DTS.

File: app/boards/arm/corneish_zen/corneish_zen_v2_left_defconfig [hunk 42]
```diff
@@ -0,0 +1,93 @@
...
+CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL=y
+CONFIG_CLOCK_CONTROL_NRF_K32SRC_30PPM=y
+
+# enable display drivers
+CONFIG_ZMK_DISPLAY_WORK_QUEUE_DEDICATED=y
+CONFIG_ZMK_DISPLAY_DEDICATED_THREAD_STACK_SIZE=2048
+CONFIG_SSD1306=n
+CONFIG_IL0323=y
+CONFIG_LVGL_HOR_RES_MAX=80
+CONFIG_LVGL_VER_RES_MAX=128
```
Comment (petejohanson) [inline app/boards/arm/corneish_zen/corneish_zen_v2_left_defconfig:43]:
```suggestion
```

Zephyr automatically sets this now from the DTS.

**Review by petejohanson — CHANGES_REQUESTED** [review]
Few more items of clean up from a final pass at this.

File: app/boards/arm/corneish_zen/corneish_zen_v2_left.dts (hunk 35 above)
Comment (LOWPROKB) [reply app/boards/arm/corneish_zen/corneish_zen_v2_left.dts:46]: This one is strange.... it looks fine VS Code locally, but when I push it there is misalignment

File: app/boards/arm/corneish_zen/corneish_zen_v2_right.dts (hunk 38 above)
Comment (LOWPROKB) [reply app/boards/arm/corneish_zen/corneish_zen_v2_right.dts:47]: Same with this one

**Review by petejohanson — APPROVED** [review]
Thanks!
